openedge-plugin/src/main/java/org/sonar/plugins/openedge/foundation/CPDCallback.java

```diff
@@ -147,4 +153,16 @@ private void visitCpdNode(JPNode node) {
           node.getEndLine(), node.getEndColumn(), uncaught.getMessage());
     }
   }
+
+  private void insertFakeNode(JPNode node) {
+    List<JPNode> children = node.getDirectChildren();
+    JPNode lastSibling = children.isEmpty() ? node : children.get(children.size() - 1);
+    try {
+      TextRange range = file.newRange(node.getLine(), node.getColumn(), lastSibling.getEndLine(), lastSibling.getEndColumn());
+      cpdTokens.addToken(range, UUID.randomUUID().toString());
+    } catch (IllegalArgumentException uncaught) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1166)